frontend/src/scenes/constructor/components/QueryBlock/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FunctionComponent } from 'react';
+import style from './style.module.scss';
+import { useDispatch, useSelector } from 'react-redux';
+import { setSelectQueryRoutine, setWaiterQueryRoutine } from '../../routines';
+import { IAppState } from '../../../../common/models/store/IAppState';
+import { ITrigger } from '../../../../common/models/query/ITrigger';
+
+interface IProps {
+    id:string,
+    name:string,
+    code?:string,
+    triggers:Array<ITrigger>|[],
+    runAutomatically:boolean|undefined,
+    showConfirm:boolean|undefined
+}
+
+const QueryBlock:FunctionComponent<IProps> = ({ id, name, code, runAutomatically, showConfirm, triggers }) => {
+  const query = useSelector((state: IAppState) => state.app.qur);
+  const dispatch = useDispatch();
+  const isDataChange = (query.selectQuery.selectQueryCode === query.setNewCode
+      && query.selectQuery.runAutomatically === query.setNewRun
+      && query.selectQuery.showConfirm === query.setNewConfirm);
+  const isTriggersChange:boolean = (
+    query.selectQuery.selectQueryTriggers.every(
+      element => [...query.setNewSuccessTriggers, ...query.setNewUnSuccessTriggers].includes(element)
+          && query.selectQuery.selectQueryTriggers.length
+          === [...query.setNewSuccessTriggers, ...query.setNewUnSuccessTriggers].length
+    ));
```

Comment:
It's difficult to understand what you're trying to do here, try to make it simpler and cleaner